internal/core/src/exec/expression/Expr.h

```diff
@@ -279,6 +279,32 @@ class SegmentExpr : public Expr {
         return result;
     }
 
+    template <typename T>
+    bool
+    CanUseIndex(OpType op) const {
+        typedef std::
+            conditional_t<std::is_same_v<T, std::string_view>, std::string, T>
+                IndexInnerType;
+        if constexpr (!std::is_same_v<IndexInnerType, std::string>) {
+            return true;
+        }
+
+        using Index = index::ScalarIndex<IndexInnerType>;
+        if (op == OpType::Match) {
+            for (size_t i = current_index_chunk_; i < num_index_chunk_; i++) {
```

Comment:
need to travel all chunk ?  all index chunk of a segment is same

Reply:
OK. thx for the reminder and i'll fix this.